### ISSUE TITLE
Offer to formalize changelog entries by offering class and type rather than component field

### ISSUE THREAD

**Issue by atoulme** [open-telemetry/opentelemetry-collector-contrib#23249]
### Is your feature request related to a problem? Please describe.

The current changelog entry format sets a field for the component to which the changes apply. However, this field is free-form and doesn't allow us to formally find if the component is spelled correctly. Often, we use informal names such as "carbonreceiver".

Ideally, by using class ("receiver", "processor", etc) and type ("carbon", "awss3", ...) we can then create a changelog that is more structured and identifies precisely the component impacted by the change.

### Describe the solution you'd like

I would like to offer a non-breaking change to the chloggen tool that allows folks to choose to use component just like today, or class and type. When rendering the changelog, the class and type fields are used if the component field is left empty, and form a string such as class/type.

Example: receiver/carbon

The change is offered on this PR: https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/326

**Comment by djaglowski** [comment]
I generally like the idea, but we still have some awkwardness w/ non-component packages. I wonder if this would generalize better if we just switch to "package" or "module", either of which would apply to non-collector repos as well, and still can be programmatically validated.

**Comment by atoulme** [comment]
Is package a replacement of type or class? Or component?

**Comment by djaglowski** [comment]
I'm suggesting we would replace the "component" field with either "package" or "module", depending on what we think is the appropriate level of granularity for the changelog.

**Comment by atoulme** [comment]
I am confused. My proposal here is to keep component but add optionally a way to be formal about composing the component field by using type and class. We can rename component to anything else we'd like as well, but I think that's a separate proposal and I don't see it as something I can use to drive better changelog semantics, so I can scope changes to a single receiver for example.

The end game for me is to start having a changelog per "component", such as say `receiver/filelog` and from there decide if we can scope releases of components individually based on whether they have changelog changes.

**Comment by djaglowski** [comment]
Right, I am suggesting that adding "type" and "class" fields is not a better solution because it is unnecessarily specific to the collector. 

Whether we keep "component" or rename it to "module", we can simply add a requirement that whatever the user specifies here must actually be a module. This validation ensures the uniformity you are looking for, I think, but can be equally useful for component modules, non-component modules, and non-collector modules.

**Comment by atoulme** [comment]
OK, let's do that instead. Let me amend the proposal.
